### DISCUSSION TITLE
Submission notifications and the role "Funding coordinator"

### DISCUSSION THREAD

**Discussion by ronste** [pkp/pkp-lib#8281] [Proposals]
Hi all,

I'd like to discuss the OJS submission workflow in the context of submission notifications and automatic assignment of participants. One of our customers run into an issue which is related to the following discussions/issues I found:

- https://forum.pkp.sfu.ca/t/ojs-3-1-2-4-automatically-assigning-a-user-with-assistant-permission-to-new-submissions/59934
- https://forum.pkp.sfu.ca/t/ojs-3-3-new-submission-notification/69028/4
- https://github.com/pkp/pkp-lib/issues/7228
- https://github.com/pkp/pkp-lib/issues/6272

The issue 6272 provides (another) work-around for what I like to discuss but doesn't really solve the issue.

The customer contacted us because non of the editors of his journal received any e-mail notifications regarding new submissions to which an editor has to be assigned any more (i.e. we are talking about the e-mail containing the locale "A new article has been submitted to which an editor needs to be assigned."). The editors used to receive these mails in the past but from one day to the other it stopped.

What happened is easy enough to explain (after one fully understands the related OJS workflow): One journal manager (to be sure to have complete access to all OJS features) assigned himself to all available roles. That's all. From this point on no notifications were sent to any editors any more.
The reason OJS was not sending such notifications any more is the role "Funding coordinator". The only assistant-level role with a stage assignment of "Submission". Since the journal manager was the only member of the "Funding coordinator" group he was automatically assigned to all new submissions and none of the editors were notified.

There are of cause easy work-arounds to deal with this. However, my point is that we have a situation here were the (appropriate) action of one user indirectly and significantly influences the workflow of other users in a non-obvious way. Up to the point where new submission could be missed altogether.

I don't really have a good idea of how to deal with this. The only thing I can come up with would be an additional setting to enable/disable automatic assignment of assistant roles. But this wouldn't really address the issue I think.

Two questions:
1) Do you think this should be addressed at all?
2) If so, can you come up with better ideas?

Best wishes,
Ronald

**Comment by NateWr** [comment]
Hi @ronste, the automated assignments system has been a source of confusion for a long time. So in 3.4 we plan to remove them. As part of work on #7191, all editorial assignments will now be configured according to the section.

![editor-assignments](https://user-images.githubusercontent.com/2306629/191464964-4bd5af71-b9a2-4771-aa92-d222b16f61b2.png)

We'll bring over the automated assignments during the upgrade: if there is only one user assigned to a manager or assistant group, then they will get a checkmark by their name after upgrade. But a journal can go in and remove this if they want. It should stop what you describe from happening in the future -- unless a zealous manager assigns everyone manually!

Does this sound like it will address the problem you raise? If so, I will go ahead and lock this proposal.

**Reply by ronste** [reply]
Dear @NateWr ,

sounds good so far. But what happens if no editor is selected in the above list? Who will receive the notification that an editor needs to be assigned? Is there still such notification or is it compulsory to select at least on editor out of the list?

**Reply by NateWr** [reply]
If no editors are assigned to a new submission, all users assigned to a user group with the `ROLE_ID_MANAGER` permission are sent an email that an editor needs to be assigned.
